### PR TITLE
chore(Core/SAI): improve the error message on invalid params

### PR DESCRIPTION
**Pull request by FrancescoBorzi**

### CODE REVIEW

**Review by IntelligentQuantum — APPROVED** [review]

File: src/server/game/AI/SmartScripts/SmartScriptMgr.cpp [hunk 1]
```diff
@@ -737,7 +737,7 @@ bool SmartAIMgr::CheckUnusedActionParams(SmartScriptHolder const& e)
         uint32 value = ((uint32*)&e.action.raw)[index];
         if (value != 0)
         {
-            LOG_ERROR("sql.sql", "SmartAIMgr: Entry %d SourceType %u Event %u Action %u has unused action_param%zu with value %u, it must be 0, skipped.",
```
Comment (IntelligentQuantum) [inline src/server/game/AI/SmartScripts/SmartScriptMgr.cpp:740]: i think here we need to change event_param{} to action_param ?

File: src/server/game/AI/SmartScripts/SmartScriptMgr.cpp [hunk 2]
```diff
@@ -800,7 +800,7 @@ bool SmartAIMgr::CheckUnusedTargetParams(SmartScriptHolder const& e)
         uint32 value = ((uint32*)&e.target.raw)[index];
         if (value != 0)
         {
-            LOG_ERROR("sql.sql", "SmartAIMgr: Entry %d SourceType %u Event %u Action %u has unused target_param%zu with value %u, it must be 0, skipped.",
+            LOG_ERROR("sql.sql", "SmartAIMgr: entryorguid {} source_type {} id {} action_type {} has unused event_param{} with value {}, it must be 0, skipped.",
```
Comment (IntelligentQuantum) [inline src/server/game/AI/SmartScripts/SmartScriptMgr.cpp:803]: event_param{} to target_param ... ?

File: src/server/game/AI/SmartScripts/SmartScriptMgr.cpp [hunk 3]
```diff
@@ -737,7 +737,7 @@ bool SmartAIMgr::CheckUnusedActionParams(SmartScriptHolder const& e)
         uint32 value = ((uint32*)&e.action.raw)[index];
         if (value != 0)
         {
-            LOG_ERROR("sql.sql", "SmartAIMgr: Entry %d SourceType %u Event %u Action %u has unused action_param%zu with value %u, it must be 0, skipped.",
+            LOG_ERROR("sql.sql", "SmartAIMgr: entryorguid {} source_type {} id {} action_type {} has unused event_param{} with value {}, it must be 0, skipped.",
```
Comment (IntelligentQuantum) [inline src/server/game/AI/SmartScripts/SmartScriptMgr.cpp:740]: event_param{} to action_param

**Comment by acidmanifesto** [comment]
superceded by https://github.com/azerothcore/azerothcore-wotlk/pull/10698
Closed recommendation by shin via conversation

**Pull request closed without merging**